<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>761</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="1">
     <widget class="QPushButton" name="auto_find_page_btn">
      <property name="text">
       <string>模板匹配&amp;&amp;自动找轮廓</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QPushButton" name="cod_page_btn">
      <property name="text">
       <string>二维码字符识别</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="defect_analysis_page_btn">
      <property name="text">
       <string>缺陷分析</string>
      </property>
     </widget>
    </item>
    <item row="0" column="3">
     <widget class="QPushButton" name="auto_find_points_btn">
      <property name="text">
       <string>自动找角点</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0" colspan="4">
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="cod_page">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>二维码扫描</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="3">
            <widget class="QPushButton" name="code_detect">
             <property name="text">
              <string>检测</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>结果</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="code_img_line">
             <property name="text">
              <string>D:/1.jpg</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>图片路径</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="code_img_select_btn">
             <property name="text">
              <string>选择</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QTextEdit" name="code_result_text">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>100</height>
              </size>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>结果图像</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="RedrawLabel" name="img_lab">
             <property name="text">
              <string/>
             </property>
             <property name="scaledContents">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="3">
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>字符识别扫描</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="3" column="1">
            <widget class="QTextEdit" name="char_result_text">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>100</height>
              </size>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>结果图像</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="RedrawLabel" name="char_img_lab">
             <property name="text">
              <string/>
             </property>
             <property name="scaledContents">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>结果</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="char_img_select_btn">
             <property name="text">
              <string>选择</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QPushButton" name="number_detect">
             <property name="text">
              <string>数字检测</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="char_img_line">
             <property name="text">
              <string>D:/charer/1.png</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>图片路径</string>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="3">
            <widget class="QPushButton" name="english_detect">
             <property name="text">
              <string>字母检测</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="auto_find_page">
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="1" column="0" colspan="4">
         <widget class="QStackedWidget" name="stackedWidget_2">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="page">
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="0" column="0">
             <widget class="QGroupBox" name="groupBox_3">
              <property name="title">
               <string>highpass查找</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_6">
               <item row="0" column="0">
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>图像位置</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="highpass_img_line"/>
               </item>
               <item row="0" column="2">
                <widget class="QPushButton" name="highpass_img_btn">
                 <property name="text">
                  <string>选择</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="3">
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>结果</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="4" rowspan="2">
                <widget class="QTextEdit" name="highpass_text"/>
               </item>
               <item row="1" column="0" colspan="3">
                <widget class="ShotLabel" name="source_img_lab">
                 <property name="minimumSize">
                  <size>
                   <width>500</width>
                   <height>500</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>500</width>
                   <height>500</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>图像展示</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0" rowspan="2" colspan="3">
                <widget class="QLabel" name="result_img_lab">
                 <property name="text">
                  <string>结果图</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="4">
                <widget class="QPushButton" name="highpass_detect_btn">
                 <property name="text">
                  <string>检测</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="4">
                <widget class="QLabel" name="result_img_lab_2">
                 <property name="text">
                  <string>裁剪区</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_2">
           <layout class="QGridLayout" name="gridLayout_8">
            <item row="0" column="0">
             <widget class="QGroupBox" name="groupBox_4">
              <property name="title">
               <string>模板匹配</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_9">
               <item row="0" column="0">
                <widget class="QLabel" name="label_9">
                 <property name="text">
                  <string>图像</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="model_img_line"/>
               </item>
               <item row="0" column="2" colspan="2">
                <widget class="QPushButton" name="model_img_select_btn">
                 <property name="text">
                  <string>选择</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_10">
                 <property name="text">
                  <string>模型</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1" colspan="2">
                <widget class="QLineEdit" name="model_line"/>
               </item>
               <item row="1" column="3">
                <widget class="QPushButton" name="model_select_btn">
                 <property name="text">
                  <string>选择</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0" colspan="5">
                <widget class="QLabel" name="model_source_lab">
                 <property name="text">
                  <string>原图</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="0" colspan="5">
                <widget class="QLabel" name="model_result_lab">
                 <property name="text">
                  <string>结果</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="5">
                <widget class="QPushButton" name="model_detect">
                 <property name="text">
                  <string>检测</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="5">
                <widget class="QLabel" name="model_contour_lab">
                 <property name="text">
                  <string>模型轮廓</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="4" colspan="2">
                <spacer name="horizontalSpacer_7">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>338</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="0" column="4" colspan="2">
                <spacer name="horizontalSpacer_6">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>338</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_3">
           <layout class="QGridLayout" name="gridLayout_10">
            <item row="0" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>图像</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="create_model_img_line"/>
            </item>
            <item row="0" column="2">
             <widget class="QPushButton" name="create_model_select_btn">
              <property name="text">
               <string>选择</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="3">
             <widget class="ShotLabel" name="create_model_lab">
              <property name="text">
               <string>图像</string>
              </property>
             </widget>
            </item>
            <item row="2" column="3">
             <widget class="QPushButton" name="create_btm">
              <property name="text">
               <string>生成</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="highpass_page_btn">
          <property name="text">
           <string>轮廓查找</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="detect_model_page_btn">
          <property name="text">
           <string>模板匹配</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="create_model_page_btn">
          <property name="text">
           <string>模板生成</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="auto_find_points_page">
       <layout class="QGridLayout" name="gridLayout_12">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_5">
          <property name="title">
           <string>轮廓找点法</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_11">
           <item row="0" column="3">
            <widget class="QPushButton" name="auto_find_points_detect">
             <property name="text">
              <string>检测</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>结果</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="auto_find_points_img_line">
             <property name="text">
              <string>D:/微信图片_20211222215801.jpg</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>图片路径</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="auto_find_points_select_btn">
             <property name="text">
              <string>选择</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QTextEdit" name="auto_find_points_result_text">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>100</height>
              </size>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>结果图像</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="RedrawLabel" name="auto_find_points_img_lab">
             <property name="text">
              <string/>
             </property>
             <property name="scaledContents">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <spacer name="horizontalSpacer_8">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="3">
            <spacer name="horizontalSpacer_9">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RedrawLabel</class>
   <extends>QLabel</extends>
   <header>redraw_label.h</header>
  </customwidget>
  <customwidget>
   <class>ShotLabel</class>
   <extends>QLabel</extends>
   <header>shot_label.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
